infra/helper.py

```diff
@@ -713,10 +716,23 @@ def build_fuzzers_impl(  # pylint: disable=too-many-arguments,too-many-locals,to
 
 def build_fuzzers(args):
   """Builds fuzzers."""
+  if args.engine == 'centipede':
+    # A hack to place target binary (with san) into a subdir of project.
+    project_name = args.project.name
+    args.project.name += f'/{args.project.name}_{args.sanitizer}'
+    build_fuzzers_impl(args.project,
+                       args.clean,
+                       args.engine,
+                       args.sanitizer,
+                       args.architecture,
+                       args.e,
+                       args.source_path,
+                       mount_path=args.mount_path)
+    args.project.name = project_name
   return build_fuzzers_impl(args.project,
                             args.clean,
                             args.engine,
-                            args.sanitizer,
+                            'none',
```

Comment:
This will break all other engines right? 

We need to only set this to 'none' in the centipede case.

Reply:
Ah yes, I had the right implementation but worried `args.project.name = project_name` may have side-effects.
I will swap back the two `build_fuzzers_impl` calls.

Reply:
Would it be a good idea to make a special case in the definition of [`_get_out_dir`](https://github.com/google/oss-fuzz/blob/8f556fe29bae30b767c8efddfd53934d25358a21/infra/helper.py#L432)?
For example, give `Project` a new attribute `subdir`, was will be created as a subdir in the project's `out/`.

The benefit is that we don't have to manually change the `out` dir twice for `Centipede` at [build_fuzzers_impl](https://github.com/google/oss-fuzz/blob/53a5887ad8275ac8b7957fbbe5e9b9e1a6ccbc1a/infra/helper.py#L652) and [check_build](https://github.com/google/oss-fuzz/blob/53a5887ad8275ac8b7957fbbe5e9b9e1a6ccbc1a/infra/helper.py#L778).
The downside is that the object `Project` might not need to know this detail about `subdir`.